### ISSUE TITLE
OctalPermissionsRule should support valid 4-digit octal permissions that don't start with '0'

### ISSUE THREAD

**Issue by reedloden** [willthames/ansible-lint#161]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible 2.1.0.0
ansible-lint 2.7.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

OctalPermissionsRule incorrectly complains about correct octal permissions. `2755` is a perfectly valid permission that uses setgid. `755` is not (should be `0755` in that case). This rule should warn when people don't include a valid 4-digit octal permission, not enforcing the zero pad.

# Actual Behaviour (Bug report only)

```
- name: setup directory (setgid)
  file: >
    state=directory
    dest=/foo/bar/log
    owner=user
    group=logs
    mode=2755
```

results in:
```
[ANSIBLE0009] Octal file permissions must contain leading zero
Task/Handler: setup directory (setgid)
```

**Comment by willthames** [comment]
No, you should use 02755 (the zero isn't to pad, it's to enforce that the number is octal)
